### PR TITLE
Websocket pr

### PR DESCRIPTION
**Pull request by AmodChaubey1110**
Exception Handling and Websocket2.0

### CODE REVIEW

**Comment by pratikbhalodiya** [comment]
it looks like worked for m

File: config/constant.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const CONSTANTS = {
+	websocketURL: 'ws://smartapisocket.angelone.in/smart-stream',
+	testingURL: 'ws://mds-uat.angelone.in/smart-stream',
```
Comment (moizm89) [inline config/constant.js:3]: please remove testing URL

File: example/test.js [hunk 2]
```diff
@@ -1,130 +1,131 @@
-let { SmartAPI, WebSocketClient } = require("smartapi-javascript");
+let { SmartAPI, WebSocketClient, WebSocketV2 } = require('../lib');
 
 let smart_api = new SmartAPI({
-    api_key: "smartapi_key",    // PROVIDE YOUR API KEY HERE
-    // OPTIONAL : If user has valid access token and refresh token then it can be directly passed to the constructor 
-    // access_token: "YOUR_ACCESS_TOKEN",
-    // refresh_token: "YOUR_REFRESH_TOKEN"
+	// api_key: 'API_KEY',
```
Comment (moizm89) [inline example/test.js:4]: why is API key commented?

File: example/test.js [hunk 3]
```diff
@@ -1,130 +1,131 @@
...
-    }).then((data) => {
-        console.log("PROFILE::", data)
-    })
-    .catch(ex => {
-        console.log("EX::", ex)
-    })
+// // If user does not have valid access token and refresh token then use generateSession method
+
+// }
+// smart_api
```
Comment (moizm89) [inline example/test.js:13]: please remove the commented code if not needed

File: package.json [hunk 4]
```diff
@@ -1,24 +1,25 @@
...
-    "address": "^1.1.2",
-    "atob": "^2.1.2",
-    "axios": "^0.20.0",
-    "pako": "^2.0.0",
-    "public-ip": "^4.0.2",
-    "querystring": "^0.2.0",
-    "ws": "^7.4.0"
-  }
+	"name": "smartapi-javascript",
+	"version": "1.0.14",
```
Comment (moizm89) [inline package.json:3]: will the version change due to new release?

File: lib/websocket_client.js [hunk 5]
```diff
@@ -1,89 +1,119 @@
...
-          apikey = params.apikey || null;
-          feedtype = params.feedtype || null;
+	let ws = null,
+		clientcode = params.clientcode || null,
+		jwttoken = params.jwttoken || null,
+		apikey = params.apikey || null;
+	feedtype = params.feedtype || null;
 
-     let url = params.url || "wss://smartapisocket.angelbroking.com/websocket?jwttoken=" + jwttoken + "&&clientcode=" +clientcode+ "&&apikey=" +apikey;
+	let url =
```
Comment (moizm89) [inline lib/websocket_client.js:19]: why changes are made to this file?

File: config/constant.js (hunk 1 above)
Comment (AmodChaubey1110) [reply config/constant.js:3]: Done

File: lib/websocket_client.js (hunk 5 above)
Comment (AmodChaubey1110) [reply lib/websocket_client.js:19]: Changed happend due to visual studio formatter , it won't affect the code flow

File: example/test.js (hunk 3 above)
Comment (AmodChaubey1110) [reply example/test.js:13]: comments were to understand the required flow

File: example/test.js (hunk 2 above)
Comment (AmodChaubey1110) [reply example/test.js]: handled now, uncommented api key

File: package.json (hunk 4 above)
Comment (AmodChaubey1110) [reply package.json]: Updated with new version, to change version with new release

**Review by moizm89 — APPROVED** [review]

**Pull request merged**